Backend/logout.php

```diff
@@ -0,0 +1,27 @@
+<?php
+header('Content-Type: application/json;charset=utf-8');
+
+$username   = $_POST['username'];
+$sessionkey = $_POST['sessionkey'];
+
+$db = new mysqli('localhost', 'XXX', 'XXX', 'XXX');
+if ($db->connect_errno > 0) {
+    die(json_encode(array('authentication' => 'false', 'error' => 'Unable to connect to database [' . $db->connect_error . ']')));
+}
+if (! isset($_POST['username'])) {
+    die(json_encode(array('authentication' => 'false', 'error' => 'username not set')));
+}
+
+include 'checkSessionkey.php';
+$json = json_encode(array('authentication' => 'false', 'error' => 'Authentication failure'));
+
+if ($check) {
+    $stmt = $db->prepare("UPDATE users SET session_timeout=? WHERE user_name=?");
+    $time_now = date('Y-m-d H:i:s', time());
+    $stmt->bind_param('ss', $time_now, $username);
+    $stmt->execute();
+    $stmt->close();
+    $json = json_encode(array('response' => 'successful'));
```

Comment:
You do not really need to create the date in php (which is of course not wrong).
You could also have written your query like this: "UPDATE users SET session_timeout=NOW() WHERE user_name =?"
This way you do not need to bind the date as a parameter, which seems a little more elegant to me (However I don't expect you to change this ;)).

Reply:
Does the NOW() function converts automatically to the Datetime-type?

Reply:
Yes, it does